core/trino-spi/src/main/java/io/trino/spi/connector/InMemoryRecordSet.java

```diff
@@ -292,7 +293,7 @@ private static long sizeOf(List<?> record)
             else if (value instanceof Boolean) {
                 completedBytes++;
             }
-            else if (value instanceof Number) {
+            else if (value instanceof Number || value instanceof LongTimestampWithTimeZone) {
```

Comment:
LongTimestampWithTimeZone is more than 8 bytes

Reply:
@findepi What would be the best way to get the size of LongTimestampWithTimeZone?

Reply:
discussed offline; 12 seems right here